Yosemite/Yosemite/Stores/InAppPurchaseStore.swift

```diff
@@ -251,11 +266,16 @@ private extension InAppPurchaseStore {
         assert(listenTask == nil, "InAppPurchaseStore.listenForTransactions() called while already listening for transactions")
 
         listenTask = Task.detached { [weak self] in
+            guard let self else {
+                return
+            }
             for await result in Transaction.updates {
                 do {
-                    try await self?.handleCompletedTransaction(result)
+                    // Wait until the purchase finishes
+                    _ = await self.pauseTransactionListener.values.contains(false)
```

Comment:
I think that is fine. Since we aren't using the App Store Server API v2 yet, we aren't really sending "a transaction" to the API. We send the app receipt, which might not have even changed between requests, and use Apple's `verifyReceipt` endpoint response to get the list of transactions.

I think dealing with potential duplicate requests is something we need to live with, and this PR doesn't change that. It only ensures that if a duplicate transaction happens, it happens in the right order.

Reply:
Thanks for your answer @koke. It's a pity that we cannot prevent at the moment sending duplicate requests in a simple way for now, but we can live with that.